JASP-Engine/JASP/R/ttestindependentsamples.R

```diff
@@ -321,80 +321,59 @@ TTestIndependentSamples <- function(dataset = NULL, options, perform = "run",
 									d <- .clean(as.numeric((ms[1] - ms[2]) / sdPooled))
-								}	else if (options$effectSizesCheckbox == "glassD") {
+								}	else if (options$effectSizesType == "glassD") {
 									d <- .clean(as.numeric((ms[1] - ms[2]) / sds[2]))
 									# Should give feedback on which data is considered 2.
-								} else if (options$effectSizesCheckbox == "hedgesG") {
-									hedges_g <- .clean(as.numeric((ms[1] - ms[2]) / sdPooled)) # same as Cohen's d so far
-									# correction_J <- function(a) { gamma(a / 2) / (sqrt(a/2) * (a - 1) / 2) } # supposedly; use approx.
-									d <- 1 - 3 / (4 * (ns[1] + ns[2]) - 9) * hedges_g # less biased / corrected version
+								} else if (options$effectSizesType == "hedgesG") {
+									a <- sum(ns) - 2
+									correction_J <- gamma(a / 2) / (sqrt(a / 2) * gamma((a - 1) / 2)) # supposedly; use approx.
+									d <- .clean(as.numeric((ms[1] - ms[2]) / sdPooled)) * correction_J # less biased / corrected version
 								}
-								
+
 							}
-							sed <-  .clean((as.numeric(r$estimate[1]) - as.numeric(r$estimate[2]))/stat)
+							sed <-  .clean((as.numeric(r$estimate[1]) - as.numeric(r$estimate[2])) / stat)
 							confIntEffSize <- c(0,0)
+
 							if (wantsConfidenceEffSize){
-							  # Taken from effsize package by Marco Torchiano, v0.7.4
-							  # Using the non-central t distributions for computing the confidence interval.
-							  # https://cran.r-project.org/web/packages/effsize/index.html
-							  # Same result as MBESS package by Ken Kelley, v4.4.3
-							  # Note that the confidence interval is the same regardless of pooling s.d.'s or (Hedges') correction.
-							  alphaLevels <- sort( c( (1-ciEffSize), ciEffSize ) )
-							  
-							  if (direction == "two.sided") {
-							    alphaLevels[1] <- (1-ciEffSize) / 2
-							    alphaLevels[2] <- (ciEffSize + 1) / 2
-							  } 
-							  
-							  st <- max(0.1,abs(stat))
-							  
-							  end1 <- stat
-							  while( pt(q = stat, df = df, ncp = end1) > alphaLevels[1]){
-							    end1 <- end1 + st
-							  }
-							  ncp1 <- uniroot(function(x) alphaLevels[1] - pt(q = stat, df = df, ncp = x),
-							                  c(2 * stat - end1, end1))$root
-				
-							  end2 <- stat
-							  while( pt(q = stat, df = df, ncp = end2) < alphaLevels[2]){
-							    end2 <- end2 - st
-							  }
-							  ncp2 <- uniroot(function(x) alphaLevels[2] - pt(q = stat, df = df, ncp = x),
-							                  c(end2, 2 * stat - end2))$root
-							  
-							  confIntEffSize = sort(c(ncp1*sqrt(1/ns[1]+1/ns[2]),  ncp2*sqrt(1/ns[1]+1/ns[2]) ))[order(c(1-ciEffSize, ciEffSize ))]
-							  
-							  if (direction == "greater") {
-							    confIntEffSize[2] <- Inf
-							  } else if (direction == "less") 
-							    confIntEffSize[1] <- -Inf
-							  
-							  confIntEffSize <- sort(confIntEffSize)
+								# From MBESS package by Ken Kelley, v4.6
+								dfEffSize <-  ifelse(effSize == "glass", ns[2]-1, df)
```

Comment:
This works well here, but be careful with ifelse in general, as it leads to weird behaviour.